app/data_pipeline.py

```diff
@@ -194,33 +195,95 @@ def index_delete(
+        }
+        file_path_indexes = []
+        if date:
+            date = datetime.strptime(date,"%m/%d/%y")
+            for i in range(len(index_object["file_path"])):
+                created_date_str = index_object["date"][i]
+                created_date = datetime.strptime(created_date_str,"%y-%m-%d")
+                if created_date >= date:
+                        file_path_indexes.append(i)
+                        try:
+                            filtered_index["file_path"].append(index_object["file_path"][i])
+                            filtered_index["file_sentence_amount"].append(index_object["file_sentence_amount"][i])
+                            filtered_index["date"].append(index_object["date"][i])
+                        except FileNotFoundError as e:
+                            raise FileExistsError(f"Index file could not be found for filtering!: {e}")
+                        
+            for index in file_path_indexes:
+                try:
+                    sentence_start = sum(sum(page_sentences) for page_sentences in index_object["file_sentence_amount"][:index])
+                    sentence_end =  sentence_start + sum(index_object["file_sentence_amount"][index])
+
+                    filtered_index["sentences"].extend(index_object["sentences"][sentence_start:sentence_end])
+                    filtered_index["embeddings"] = np.vstack((index_object["embeddings"][sentence_start:sentence_end],filtered_index["embeddings"]))
+
+                except FileNotFoundError as e:
+                    raise FileExistsError(f"Index file could not be found for filtering!: {e}")
+            else:
+                return filtered_index
+        else:
+            return index_object
+
+    def preprocessed_query(self, query):
+        return self.cf.query_generation(query=query)
+
     def search_index(
             self,
-            user_query: np.ndarray,
+            user_query: np.ndarray
     ):
-        query_vector = self.ef.create_vector_embedding_from_query(query=user_query)
-        _, I = globals.index.search(query_vector, 5)
-        widen_sentences = self.widen_sentences(window_size=1, convergence_vector=I[0])
-        context = f"""Context1: {widen_sentences[0]}
-        Context2: {widen_sentences[1]}
-        Context3: {widen_sentences[2]}
-        Context4: {widen_sentences[3]}
-        Context5: {widen_sentences[4]}
-        """
-        resources = self.extract_resources(convergence_vector=I[0])
+        split_queries = user_query.split('\n')
+        globals.og_query = split_queries[0]
+        index_list = []
+        for query in split_queries:
+            if(len(query) < 10):
+                continue
+            else:
+                query_vector = self.ef.create_vector_embedding_from_query(query=query)
+                _, I = globals.index.search(query_vector, 3)
+                index_list.extend(I[0])
+        unique_index_list = list(set(index_list))
```

Comment:
why did you define this as list first, then change it to set then rechange it to list again? define this as set first, add the elements then change it to list